gril/grilunsol.c

```diff
@@ -39,6 +39,32 @@
 #include "common.h"
 #include "grilunsol.h"
 
+int g_ril_unsol_parse_connected(GRil *gril, const struct ril_msg *message)
+{
+	struct parcel rilp;
+	int size;
+	int version = RIL_VERSION_UNSPECIFIED;
+
+	DBG("");
+
+	g_ril_init_parcel(message, &rilp);
+
+	if ((rilp.offset + 2 * sizeof(int32_t)) > rilp.size) {
```

Comment:
I prefer these days to have a final check of the malformed flag *after* reading all data, something like:

		...
		version = parcel_r_int32(&rilp);

		if (rilp.malformed) {
			ofono_error("%s: malformed parcel received", __func__);
			version = RIL_VERSION_UNSPECIFIED;
		}